libs/ballot-interpreter-nh-next/src/layout.rs

```diff
@@ -0,0 +1,112 @@
+use itertools::Itertools;
+use serde::Serialize;
+
+use crate::{
+    ballot_card::BallotSide,
+    election::{ContestId, GridLayout, GridPosition, OptionId},
+    geometry::{GridUnit, PixelPosition, Point, Rect},
+    timing_marks::TimingMarkGrid,
+};
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct InterpretedContestOptionLayout {
+    option_id: OptionId,
+    bounds: Rect,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct InterpretedContestLayout {
+    contest_id: ContestId,
+    bounds: Rect,
+    options: Vec<InterpretedContestOptionLayout>,
+}
+
+fn build_option_layout(
+    grid: &TimingMarkGrid,
+    grid_position: &GridPosition,
+) -> Option<InterpretedContestOptionLayout> {
+    // Option bounding box parameters
+    // TODO make these configurable in the election definition
+    let column_offset: i32 = -9;
+    let row_offset: i32 = -1;
+    let width: GridUnit = 10;
+    let height: GridUnit = 2;
+
+    let clamp_row = |row: i32| -> GridUnit {
+        return row.max(0).min(grid.geometry.grid_size.height as i32 - 1) as GridUnit;
+    };
+    let clamp_column = |column: i32| -> GridUnit {
+        return column.max(0).min(grid.geometry.grid_size.width as i32 - 1) as GridUnit;
+    };
+
+    let bubble_location = grid_position.location();
+
+    let top_left_grid_point: Point<GridUnit> = Point::new(
+        clamp_column(bubble_location.column as i32 + column_offset),
+        clamp_row(bubble_location.row as i32 + row_offset),
+    );
+    let bottom_right_grid_point: Point<GridUnit> = Point::new(
+        clamp_column(bubble_location.column as i32 + column_offset + width as i32),
+        clamp_row(bubble_location.row as i32 + row_offset + height as i32),
+    );
+
+    let top_left_subpixel_point = grid.point_for_location(
+        top_left_grid_point.x as GridUnit,
+        top_left_grid_point.y as GridUnit,
+    )?;
+    let bottom_right_subpixel_point = grid.point_for_location(
+        bottom_right_grid_point.x as GridUnit,
+        bottom_right_grid_point.y as GridUnit,
+    )?;
+    let top_left_pixel_point = Point::new(
+        top_left_subpixel_point.x as PixelPosition,
+        top_left_subpixel_point.y as PixelPosition,
+    );
+    let bottom_right_pixel_point = Point::new(
+        bottom_right_subpixel_point.x as PixelPosition,
+        bottom_right_subpixel_point.y as PixelPosition,
+    );
```

Comment:
Makes me wish for a `Point::round`. Then you could do this:

```suggestion
    let top_left_pixel_point = top_left_subpixel_point.round();
    let bottom_right_pixel_point = bottom_right_subpixel_point.round();
```

Actually, that's not hard to do:

```rust
impl Point<SubPixelUnit> {
    pub fn round(self) -> Point<PixelPosition> {
        Point::new(
            self.x.round() as PixelPosition,
            self.y.round() as PixelPosition,
        )
    }
}
```

Reply:
Yes, was thinking of something like this! Thanks!